contrib/initramfs/zfsunlock

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -eu
+echo "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfs_fs_name ] && [ ! -e /run/zfs_unlock_complete ]; do
+	sleep 1;
+done
+if [ -e /run/zfs_unlock_complete ]; then
+	exit 0
+fi
+echo
+echo "Unlocking encrypted ZFS pools ..."
+echo "Enter the password or press Ctrl-C to exit into shell."
+echo
+zfs_fs_name=""
+fs_password_accepted=false
+while true; do
+	while [ "$fs_password_accepted" = "true" ] && [ "$(cat /run/zfs_fs_name)" = "$zfs_fs_name" ] && [ ! -e /run/zfs_unlock_complete ]; do
+		sleep 1;
+        done
+	if [ -e /run/zfs_unlock_complete ]; then
+		break
+	fi
+	zfs_fs_name=$(cat /run/zfs_fs_name)
+	zfs_console_askpwd_cmd=$(cat /run/zfs_console_askpwd_cmd)
+	systemd-ask-password "Encrypted ZFS password for ${zfs_fs_name}:" | \
+		/sbin/zfs load-key "$zfs_fs_name"
+	if [ "$(/sbin/zfs get -H -ovalue keystatus "$zfs_fs_name" 2> /dev/null)" = "available" ]; then
+		fs_password_accepted=true
+		echo "Password for $zfs_fs_name accepted."
+		zfs_console_askpwd_pid=$(ps a -o pid= -o args | grep -v grep | grep "$zfs_console_askpwd_cmd" | cut -d ' ' -f3 | sort -n | head -n1)
+                echo "Resuming boot sequence."
+		echo "Please reconnect in a while."
+		if [ -n "$zfs_console_askpwd_pid" ]; then
```

Comment:
For clarity, I would put the lookup of the PID and the killing of it next to each other, and the echos next to each other, rather than mixing them.